web/src/components/njwds-extended/HorizontalStepper.tsx

```diff
@@ -96,39 +119,82 @@ export const HorizontalStepper = (props: Props): ReactElement => {
     }
   };
 
-  const getBottomMargin = (): string => {
-    return isTabletAndUp ? "margin-bottom-4" : "";
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>, index: number): void => {
+    if (event.key === "Enter" || event.key === " ") {
+      setFocusStep(props.currentStep);
+      divRefs.current[props.currentStep]?.focus();
+      props.onStepClicked(index);
+    }
+
+    if (event.key === "ArrowRight") {
+      if (focusStep === props.steps.length - 1) {
+        divRefs.current[0]?.focus();
+        setFocusStep(0);
+      } else {
+        divRefs.current[focusStep + 1]?.focus();
+        setFocusStep((prevFocusStep) => prevFocusStep + 1);
+      }
+    }
+
+    if (event.key === "ArrowLeft") {
+      if (focusStep === 0) {
+        divRefs.current[props.steps.length - 1]?.focus();
+        setFocusStep(props.steps.length - 1);
+      } else {
+        divRefs.current[focusStep - 1]?.focus();
+        setFocusStep((prevFocusStep) => prevFocusStep - 1);
+      }
+    }
+
+    if (event.key === "Tab") {
+      setFocusStep(props.currentStep);
+    }
   };
 
+  const divRefs = useRef<(HTMLDivElement | null)[]>([]);
+
+  useEffect(() => {
+    if (divRefs.current[props.currentStep] && !props.selectedStepHasErrors) {
+      scrollToTopOfElement(divRefs.current[props.currentStep], { focusElement: true });
+    }
+  }, [props.currentStep, props.selectedStepHasErrors]);
+
   return (
     <>
       <div className="horizontal-step-indicator display-block">
         <div className="usa-step-indicator usa-step-indicator--counters-sm">
-          <ol className={`usa-step-indicator__segments ${getBottomMargin()}`}>
+          <div className={`usa-step-indicator__segments stepper-wrapper`} role={"tablist"}>
             {props.steps.map((step: StepperStep, index: number) => {
               return (
-                <li
+                <div
                   key={`${step.name}-${index}`}
                   className={
                     `border-bottom-2px ${getBorderColor(index)} cursor-pointer ` +
                     `usa-step-indicator__segment usa-step-indicator__segment${getCSSClassColor(index)}`
                   }
-                  aria-hidden
                   data-num={getIcon(index)}
                   data-state={determineState(index)}
-                  onClick={(): void => props.onStepClicked(index)}
                   data-testid={`stepper-${index}`}
+                  onClick={(): void => props.onStepClicked(index)}
+                  onKeyDown={(e) => handleKeyDown(e, index)}
+                  role="tab"
+                  tabIndex={index === props.currentStep ? 0 : -1}
+                  aria-label={`Formation Stepper Navigation: ${step.name} Step, State: ${determineAriaState(
```

Comment:
[sand] - I think its quite likely this should be in content and will require a follow up.

Not addressed here because scope has already creeped a lot. 

(Also this one is non-trivial as it requires some fun content token interpolation for the variables here)